x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.test.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { shallow, ShallowWrapper } from 'enzyme';
+
+import { EuiComboBox, EuiFieldText } from '@elastic/eui';
+
+import { AttributeSelector } from './attribute_selector';
+import { ANY_AUTH_PROVIDER, ANY_AUTH_PROVIDER_OPTION_LABEL } from './constants';
+
+const handleAttributeSelectorChange = jest.fn();
+const handleAttributeValueChange = jest.fn();
+const handleAuthProviderChange = jest.fn();
+
+const baseProps = {
+  attributeName: 'An Attribute',
+  attributeValue: 'Something',
+  attributes: ['a', 'b', 'c'],
+  availableAuthProviders: ['ees_saml', 'kbn_saml'],
+  selectedAuthProviders: ['ees_saml'],
+  elasticsearchRoles: ['whatever'],
+  multipleAuthProvidersConfig: true,
+  disabled: false,
+  handleAttributeSelectorChange,
+  handleAttributeValueChange,
+  handleAuthProviderChange,
+};
+
+describe('AttributeSelector', () => {
+  it('renders', () => {
+    const wrapper = shallow(<AttributeSelector {...baseProps} />);
+
+    expect(wrapper.find('[data-test-subj="attributeSelector"]')).toBeDefined();
+  });
+
+  it('renders disabled panel with className', () => {
+    const wrapper = shallow(<AttributeSelector {...baseProps} disabled />);
+
+    expect(wrapper.find('[data-test-subj="attributeSelector"]').prop('className')).toEqual(
+      'euiPanel--disabled'
+    );
+  });
+
+  describe('Auth Providers', () => {
+    const findAuthProvidersSelect = (wrapper: ShallowWrapper) =>
+      wrapper.find('[data-test-subj="authProviderSelect"]');
+
+    it('will not render if "availableAuthProviders" prop has not been provided', () => {
+      const wrapper = shallow(
+        <AttributeSelector {...baseProps} availableAuthProviders={undefined} />
+      );
+
+      expect(wrapper.find('[data-test-subj="authProviderSelect"]')).toHaveLength(0);
+    });
+
+    it('handles fallback props', () => {
```

Comment:
FWIW, I don't really understand what this means... does it mean like .... if attributeValue and selectedAuthProviders aren't provided then it populates a different set of select options?

Reply:
Sorry, I didn't know of a better way to name this. I added this for test coverage for the lines that have fallback props. Perhaps "default" props would be a better name but am open to suggestions:

```JavaScript
export const AttributeSelector: React.FC<Props> = ({
  attributeName,
  attributeValue = '', // this
  attributes,
  availableAuthProviders,
  selectedAuthProviders = [ANY_AUTH_PROVIDER], // and this
  ...
```